zerver/models.py

```diff
@@ -2374,7 +2377,7 @@ def get_user_profile_by_api_key(api_key: str) -> UserProfile:
 
     return user_profile
 
-def get_user_by_delivery_email(email: str, realm: Realm) -> UserProfile:
+def get_user_by_delivery_email(email: str, realm: Optional[Realm]=None) -> UserProfile:
```

Comment:
Don't make this optional. It's a key function used in many places and we don't want to mess with it.